src/databricks/labs/ucx/framework/crawlers.py

```diff
@@ -110,7 +110,7 @@ def _crawl(self) -> Iterable[Result]:
             Iterable[Result]: Records that capture the results of crawling the environment.
         """
 
-    def _snapshot(self, fetcher: ResultFn, loader: ResultFn) -> list[Result]:
+    def _snapshot(self, fetcher: ResultFn, loader: ResultFn, *, force_refresh: bool) -> list[Result]:
```

Comment:
I slightly doubt the naming `force_refresh`. Why not just `refresh`; what does "force" add? Otherwise, maybe it should be `force_fetch` as that is forced. My preference goes to `crawler.snapshot(refresh=True)`.

Note that in dbt they use "full refresh" as they support incremental refreshing. In our case a "refresh" is always "full"

Reply:
Naming is hard. ;)

The `force_` prefix is present to make the purpose unambiguous. Without this prefix it could be misleading: `refresh=False`, at face value, could be read to indicate that it's not allowed to perform a crawl. (However it will if the underlying table doesn't exist or is empty.) As such the name 'force_refresh' is intended to convey that if it's `True` we definitely will, but if it's `False` that doesn't mean we won't.

I don't like the idea of using `fetch` in the name: use of the term fetch is inconsistent, and internally within the class it means "load the snapshot from the table" instead of "perform a crawl" as you intended. For now cleaning up the use of this term is tech debt and outside the scope of this PR.

Does this make sense?

Reply:
Fair enough, I agree with your reasoning. Naming is hard indeed :/